packages/types/src/primitive/Method.ts

```diff
@@ -19,8 +25,12 @@ interface DecodeMethodInput {
 interface DecodedMethod extends DecodeMethodInput {
   argsDef: ArgsDef;
   meta: FunctionMetadataV5;
+  method?: string;
+  section?: string;
 }
 
+export type MetaLike = FunctionMetadataV5 | MethodFunction | Metadata;
```

Comment:
I'm still not convinced about MethodFunction.

And as you proposed in Riot, i would maybe go for removing MethodFunction in the whole codebase.